src/components/LineChart/LineChart.jsx

```diff
@@ -96,18 +195,100 @@ export default class LineChart extends PureComponent {
     this.lines = this.g.append('g').classed('lines-group', true);
     this.circles = this.g.append('g').classed('circles-group', true);
 
+    // container for showing the x highlighte date indicator
+    this.highlightDate = this.g.append('g').attr('class', 'highlight-date');
+    this.highlightDate.append('line')
+      .attr('x1', 0)
+      .attr('x2', 0)
+      .attr('y1', 0)
+      .attr('y2', innerHeight + 3)
+      .attr('class', 'highlight-ref-line');
+
+    // container for showing the highlighted line
+    this.highlightLine = this.g.append('g').attr('class', 'highlight-line');
+    this.highlightLine.append('rect').attr('class', 'series-overlay');
+    this.highlightLine.append('g');
+
+    // add in mouse listener -- this should be added above the lines and axes
+    const that = this;
+    this.mouseListener = this.g.append('rect')
+      .attr('class', 'mouse-listener')
+      .style('fill', '#f00')
+      .style('stroke', 'none')
+      .style('opacity', 0)
+      .on('mousemove', function mouseMoveListener() {
+        that.onMouseMove.call(that, d3.mouse(this));
+      })
+      .on('mouseout', () => this.onMouseMove(null));
+
+
     this.update();
   }
 
   /**
+   * Helper function to make components necessary for the legend
+   */
+  makeLegendComponents(series = [], annotationSeries = [], legendWidth) {
```

Comment:
would it start to make sense to pull the legend out into a separate component - if we plan on potentially using it elsewhere?